src/libraries/System.Diagnostics.Process/tests/ProcessThreadTests.cs

```diff
@@ -148,17 +149,22 @@ public async Task TestStartTimeProperty()
                 await Task.Factory.StartNew(() =>
                 {
                     p.Refresh();
-                    try
-                    {
-                        var newest = p.Threads.Cast<ProcessThread>().OrderBy(t => t.StartTime.ToUniversalTime()).Last();
```

Comment:
Was it wrongly assumed that Process.Threads contained the threads ordered with the most recently created as the `Last()` one? 

Is also possible that the property does have the threads ordered but StartNew wasn't creating a new thread but reusing an existing one as Dan mentioned.

Reply:
> Was it wrongly assumed that Process.Threads contained the threads ordered with the most recently created as the Last() one?

There are other tests running in parallel and each of them can create a new thread. Even if we disable that, the `ThreadPool` can still start a new thread on it's own. So to get 100% deterministic behavior we should IMO perform an ID match.

Reply:
> StartNew wasn't creating a new thread but reusing an existing one as Dan mentioned.

In theory CLR and Mono are sharing the same thread pool implementation, so I would expect `TaskCreationOptions.LongRunning` to always create a new thread, but I might be wrong.

@kouvel is it safe to assume that `Task.Factory.StartNew(/**/, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);` executed from an xUnit test runner will always spawn a new thread?

Reply:
Based on this:
https://github.com/dotnet/runtime/blob/27776e2fbcf865029829027bef9a272326e70105/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ThreadPoolTaskScheduler.cs#L45-L53

It looks like when thread start is supported, a new thread is created for each long-running task. The threads used for long-running tasks are not pooled currently, though they could be. Probably creating a new thread would be more explicit in intent.

Wonder if the failure is happening when thread start is not supported. In the threading tests that create new threads, the following condition is used, which appears to be the same condition as `Thread.IsThreadStartSupported`:
`[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]`

Reply:
Although using a long-running task would avoid an unhandled exception/crash on assertion failure in the new thread. `ThreadTestHelpers.CreateGuardedThread` could be used for, as in: https://github.com/dotnet/runtime/blob/27776e2fbcf865029829027bef9a272326e70105/src/libraries/System.Threading.Thread/tests/ThreadTests.cs#L621

Reply:
@kouvel thank you!

Reply:
> The threads used for long-running tasks are not pooled currently, though they could be.

TIL - I should have checked the code. @kouvel is that worth an issue, as a suggestion? Basically pull a thread from the pool if available, and trigger a replacement to start. I have no idea.

Reply:
What's done currently is simple and may be sufficient if most long-running work items rarely or never complete. If there's a need where work items would complete frequently enough, and are not short enough and need to be classified as long-running, then it may be worthwhile to pool those threads.